kolibri/core/assets/src/api-resource.js

```diff
@@ -186,7 +186,7 @@ class Model {
             this.resource.removeModel(this);
```

Comment:
at the beginning of this method, it checks ```if (!this.id)``` and reject a message if this.id is undefined.